chapter-06/inprogress/suggestions.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import "sort"
+
+// bookSuggestions are the suggested books for a given book.
+// Suggestions are the neighbour-books on the shelves.
+type bookSuggestions map[Book]bookCollection
+
+// bookCollection is a collection of books.
+type bookCollection map[Book]struct{}
```

Comment:
I think you also want have some sort of "recommendation score", because some books are probably more relevant than others.